common/persistence/sql/queue_v2.go

```diff
@@ -26,36 +26,183 @@ package sql
 
 import (
 	"context"
-	"errors"
+	"database/sql"
 	"fmt"
 
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/api/serviceerror"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
 	"go.temporal.io/server/common/persistence"
+	"go.temporal.io/server/common/persistence/sql/sqlplugin"
 )
 
-type (
-	queueV2 struct{}
+const (
+	EmptyPartition = 0
+	// pageTokenPrefixByte is the first byte of the serialized page token. It's used to ensure that the page token is
+	// not empty. Without this, if the last_read_message_id is 0, the serialized page token would be empty, and clients
+	// could erroneously assume that there are no more messages beyond the first page. This is purely used to ensure
+	// that tokens are non-empty; it is not used to verify that the token is valid like the magic byte in some other
+	// protocols.
+	pageTokenPrefixByte = 0
 )
 
-var (
-	ErrNotImplemented = errors.New("method is not implemented yet")
+type (
+	queueV2 struct {
+		logger log.Logger
+		SqlStore
+	}
+
+	QueueV2MetadataPayload struct {
+		AckLevel int64
+	}
 )
 
-// NewQueueV2 returns an implementation of [persistence.QueueV2] which always returns an error because it is not
-// implemented yet.
-func NewQueueV2() persistence.QueueV2 {
-	return &queueV2{}
+// NewQueueV2 returns an implementation of persistence.QueueV2.
+func NewQueueV2(db sqlplugin.DB,
+	logger log.Logger) (persistence.QueueV2, error) {
+
+	queue := &queueV2{
+		SqlStore: NewSqlStore(db, logger),
+		logger:   logger,
+	}
+	return queue, nil
 }
 
 func (q queueV2) EnqueueMessage(
```

Comment:
We should make this a pointer receiver here so that it's consistent with all other receivers

Reply:
Done